examples/rust/get_started/examples/05-secure-channel.rs

```diff
@@ -1,29 +1,37 @@
 // This node creates a secure channel and routes a message through it.
 
-use ockam::{Context, Result, Route, SecureChannel, Vault};
+use ockam::{Address, Context, Entity, IdentifierTrustPolicy, ProfileIdentity, Result, Route};
 use ockam_get_started::Echoer;
 
 #[ockam::node]
 async fn main(mut ctx: Context) -> Result<()> {
     // Start an Echoer worker at address "echoer"
     ctx.start_worker("echoer", Echoer).await?;
 
-    let vault = Vault::create(&ctx)?;
+    let mut bob = Entity::create(&ctx).await?;
+
+    // Connect to a secure channel listener and perform a handshake.
+    let mut alice = Entity::create(&ctx).await?;
+
+    // Bob defines a trust policy that only trusts Alice
+    let bob_trust_policy = IdentifierTrustPolicy::new(alice.identifier()?);
+
+    // Alice defines a trust policy that only trusts Bob
+    let alice_trust_policy = IdentifierTrustPolicy::new(bob.identifier()?);
 
     // Create a secure channel listener.
-    SecureChannel::create_listener(&ctx, "secure_channel_listener", &vault).await?;
+    bob.create_secure_channel_listener("bob_secure_channel_listener", bob_trust_policy)
+        .await?;
 
-    // Connect to a secure channel listener and perform a handshake.
-    let channel = SecureChannel::create(&ctx, "secure_channel_listener", &vault).await?;
+    let channel_to_bob = alice
+        .create_secure_channel("bob_secure_channel_listener", alice_trust_policy)
+        .await?;
+
+    let echoer: Address = "echoer".into();
+    let route: Route = vec![channel_to_bob, echoer].into();
 
     // Send a message to the echoer worker, via the secure channel.
-    ctx.send(
-        // route to the "echoer" worker via the secure channel.
-        Route::new().append(channel.address()).append("echoer"),
-        // the message you want echo-ed back
-        "Hello Ockam!".to_string(),
-    )
-    .await?;
+    ctx.send(route, "Hello Ockam!".to_string()).await?;
```

Comment:
Can't put the Route vec in line here without some changes to `send`, due to the way the generics are handled. When one tries, the compiler says: `cannot infer type for type parameter R`. That is the `R: Into<Route>` trait bound. Seems like there should be enough compile time information here.